### PR TITLE
SNOW-637199: add snowflake snowpark context get active session to the api for streamlit

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-637199

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   - Created a new module `context` with a `get_active_session` function that is a public API to get current active snowpark sessions
   - Added test to ensure that the behavior is as expected

### CODE REVIEW

File: tests/integ/test_context.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+
+from snowflake.snowpark.context import get_active_session
+
+
+def test_get_active_session(session):
```
Comment (sfc-gh-aalam) [inline tests/integ/test_context.py:9]: I considered adding some negative tests for no active sessions and for multiple sessions but they are already present in `tests/integ/test_session.py`.

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/context.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+from snowflake.snowpark import Session
```
Comment (sfc-gh-jdu) [inline src/snowflake/snowpark/context.py:5]: nit: You can combine these two imports to 
```
from snowflake.snowpark.session import Session, _get_active_session
```

File: src/snowflake/snowpark/context.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+from snowflake.snowpark.session import _get_active_session
+
+
+def get_active_session() -> "Session":
```
Comment (sfc-gh-jdu) [inline src/snowflake/snowpark/context.py:9]:
```suggestion
def get_active_session() -> Session:
```

Comment (sfc-gh-jdu) [reply src/snowflake/snowpark/context.py:9]: Since it's a public function, you should add docstring

File: src/snowflake/snowpark/context.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+
```
Comment (sfc-gh-jdu) [inline src/snowflake/snowpark/context.py:5]: I guess you also need a docstring for this new file (similar to https://github.com/snowflakedb/snowpark-python/blob/196cb4e9d3fd4a862bf07f55c9c16433925f3b49/src/snowflake/snowpark/udf.py#L5)

**Review by sfc-gh-jdu — APPROVED** [review]

File: src/snowflake/snowpark/context.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+from snowflake.snowpark.session import Session, _get_active_session
+
+
+def get_active_session() -> Session:
+    """Returns the current active snowpark session. Throws an error when
+    multiple sessions are active or when no sessions are created."""
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/context.py:11]: Exception in docstring can be like. We use google-style docstring. https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

```python
"""Returns the current active snowpark session. 

Raises:
    SnowparkSessionException: If there are more than one active sessions or no active sessions.
"""
```

File: src/snowflake/snowpark/context.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+"""Context module for Snowpark."""
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/context.py:5]: @sfc-gh-mabrennan could you review the text in the module and function?

**Review by sfc-gh-yixie — APPROVED** [review]

File: src/snowflake/snowpark/context.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+from snowflake.snowpark.session import Session, _get_active_session
+
+
+def get_active_session() -> Session:
+    """Returns the current active snowpark session.
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/context.py:10]: snowpark should probably be capitalized.

File: src/snowflake/snowpark/context.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+    Raises: SnowparkSessionException: If there are more than one active sessions or no active sessions.
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/context.py:12]: Change "If there are more than one active sessions or no active sessions." to "If there is more than one active session or no active sessions."

File: src/snowflake/snowpark/context.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+    Returns:
+        A :class:`Session` object for current session.
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/context.py:15]: add "the" before "current"

**Review by sfc-gh-mabrennan — COMMENTED** [review]
Thank you for writing the docstrings. They look great. I just made a couple of small suggestions.

**Comment by sfc-gh-jdu** [comment]
Sorry I should realize it earlier that we should also add `snowflake.snowpark.context` here https://github.com/snowflakedb/snowpark-python/blame/main/docs/source/_templates/module.rst#L24 so the doc can have this class and method. @sfc-gh-aalam could you fix it and also check whether you can find it in the doc? This is the instruction of building docs: https://github.com/snowflakedb/snowpark-python/tree/main/docs

**Pull request merged**